provider/aws/elbv2.go

```diff
@@ -0,0 +1,272 @@
+	client           ELBV2ClientDescreptor
+	storage          storage.Storage
+	cloudWatchCLient *CloudwatchManager
+	pricingClient    *PricingManager
+	metrics          []config.MetricConfig
+	region           string
+
+	namespace          string
+	servicePricingCode string
+}
+
+//
+type ARNManager struct {
+	Partition string
+	Service   string
+	Region    string
+	AccountID string
+	Resource  string
+}
+
+// DetectedELB define the detected AWS ELB instances
+type DetectedELBV2 struct {
+	Metric string
+	Region string
+	structs.BaseDetectedRaw
+}
+
+func ParseArn(arn string) (ARNManager, error) {
+	const (
+		arnDelimiter = ":"
+		arnSections  = 6
+		arnPrefix    = "arn:"
+
+		// zero-indexed
+		sectionPartition = 1
+		sectionService   = 2
+		sectionRegion    = 3
+		sectionAccountID = 4
+		sectionResource  = 5
+
+		// errors
+		invalidPrefix   = "arn: invalid prefix"
+		invalidSections = "arn: not enough sections"
+	)
+	if !strings.HasPrefix(arn, arnPrefix) {
+		return ARNManager{}, errors.New(invalidSections)
+	}
+	sections := strings.SplitN(arn, arnDelimiter, arnSections)
+	if len(sections) != arnSections {
+		return ARNManager{}, errors.New(invalidSections)
+	}
+
+	return ARNManager{
+		Partition: sections[sectionPartition],
+		Service:   sections[sectionService],
+		Region:    sections[sectionRegion],
+		AccountID: sections[sectionAccountID],
+		Resource:  sections[sectionResource],
+	}, nil
+
+}
+
+// TableName will set the table name to storage interface
+func (DetectedELBV2) TableName() string {
+	return "aws_elbv2"
+}
+
+// NewELBManager implements AWS GO SDK
+func NewELBV2Manager(client ELBV2ClientDescreptor, st storage.Storage, cloudWatchCLient *CloudwatchManager, pricing *PricingManager, metrics []config.MetricConfig, region string) *ELBV2Manager {
+
+	st.AutoMigrate(&DetectedELBV2{})
+
+	return &ELBV2Manager{
+		client:           client,
+		storage:          st,
+		cloudWatchCLient: cloudWatchCLient,
+		metrics:          metrics,
+		pricingClient:    pricing,
+		region:           region,
+
+		namespace:          "AWS/ApplicationELB",
+		servicePricingCode: "AmazonEC2",
+	}
+}
+
+// Detect check with ELB  instance is under utilization
+func (r *ELBV2Manager) Detect() ([]DetectedELBV2, error) {
+	log.Info("Analyze ELBV2")
+	detectedELBV2 := []DetectedELBV2{}
+
+	instances, err := r.DescribeLoadbalancers()
+	if err != nil {
+		return detectedELBV2, err
+	}
+
+	now := time.Now()
+
+	for _, instance := range instances {
+
+		log.WithField("name", *instance.LoadBalancerName).Info("check ELBV2")
+
+		price, _ := r.pricingClient.GetPrice(r.GetPricingFilterInput(), "")
+
+		for _, metric := range r.metrics {
+
+			log.WithFields(log.Fields{
+				"name":        *instance.LoadBalancerName,
+				"metric_name": metric.Description,
+			}).Debug("check metric")
+
+			period := int64(metric.Period.Seconds())
+
+			metricEndTime := now.Add(time.Duration(-metric.StartTime))
+			parsedARN, err := ParseArn(*instance.LoadBalancerArn)
```

Comment:
Do we really need this function? why are yo change the ARN resource?

Reply:
I believe yes, we need it.
This is the only way I found go get metric from cloudwatch. Here is why: 

`Dimensions: []*cloudwatch.Dimension{
					&cloudwatch.Dimension{
						Name:  awsClient.String("LoadBalancer"),
						Value: &resourseParsed,`

The path for metric we want to get in Cloudwatch UI looks like "All > ApplicationELB > Per ELB metrics", this is OK, we get there with SDK, but then we should request "LoadBalancer" with his "Name", unfortunately, Application Loadbalancers has specific kind of "Name" which we can't get anywhere except parsing arn and cut necessary substring. 

More concrete:
When we iterate over Application ELB we have next useful fields in SDK:
- LoadBalancerArn (this is full arn)
- LoadBalancerName (short name, like 'my-loadbalancer' this is visible in EC2 console)
- Type ( substring "application")

In Cloudwatch the ALB metrics stores with "LoadBalancer Name" which looks like: 'app/my-loadbalancer/12313aqwe' this substring is a part of arn. So I parced it.

Reply:
Here is AWS Docs:
https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-cloudwatch-metrics.html

Reply:
Hey  @onagorodniuk, 
With my check when you pass the only the `instance.LoadBalancerArn` it's works.
see my change:
```
...
// regx, _ := regexp.Compile(".*loadbalancer/")
			// resourseParsed := regx.ReplaceAllString(parsedARN.Resource, "")

			metricInput := cloudwatch.GetMetricStatisticsInput{
				Namespace:  &r.namespace,
				MetricName: &metric.Description,
				Period:     &period,
				StartTime:  &metricEndTime,
				EndTime:    &now,
				Dimensions: []*cloudwatch.Dimension{
					&cloudwatch.Dimension{
						Name:  awsClient.String("LoadBalancer"),
						Value: instance.LoadBalancerArn,
					},
				},
			}
...


Can you check it in your end ? 
```

Reply:
Yes, it works but cloudwatch always return zero. All ALB's is marking as not utilized.
I have recheck this again yesterday, when reiwing you comments.

Reply:
Ok I understand, but still you don't needs the `ParseArn` function

you can pass `*instance.LoadBalancerAr` to `regx.ReplaceAllString` without using `ParseArn` function.
Please see the example:

```
metricEndTime := now.Add(time.Duration(-metric.StartTime))
			// parsedARN, err := ParseArn(*instance.LoadBalancerArn)

			regx, _ := regexp.Compile(".*loadbalancer/")
			resourseParsed := regx.ReplaceAllString(*instance.LoadBalancerArn, "")

			metricInput := cloudwatch.GetMetricStatisticsInput{
				Namespace:  &r.namespace,
				MetricName: &metric.Description,
				Period:     &period,
				StartTime:  &metricEndTime,
				EndTime:    &now,
				Dimensions: []*cloudwatch.Dimension{
					&cloudwatch.Dimension{
						Name:  awsClient.String("LoadBalancer"),
						Value: &resourseParsed,
					},
				},
			}
```

Can you please check it? 
What do you think?